newIDE/app/src/ProjectCreation/LocalExamples.js

```diff
@@ -80,10 +84,12 @@ export default class LocalExamples extends Component<Props, State> {
     });
   }
 
-  _handleChangePath = (outputPath: string) =>
+  _handleChangePath = (outputPath: string) => {
```

Comment:
After moving outputPath to the parent, you don't need _handleChangePath. Instead, you call `this.props.onChangeOutputPath`. It's the parent responsibility to store it in its state.

Why does it work? Because *any time* you change the state of a component, *then* the component is re-rendered, and all children components are-rendered with the updated props (that may or may have not changed). That's the "react way of thinking", which is ver clean once you get it:
you store things in state. You can pass props to your children, including callbacs. These callbacks can updating the state. If it's done, everything that is in the component will be re-rendered ("flowing" from parents to children). Done! :) 
See https://reactjs.org/docs/thinking-in-react.html

Reply:
Oh, that makes more sense now. Thank you very much for your time explaining it to me! (I should really have checked the react Docs before messing with it 😓 )